addons/l10n_din5008/models/account_move.py

```diff
@@ -45,13 +45,21 @@ def _compute_l10n_din5008_document_title(self):
     def _compute_l10n_din5008_addresses(self):
         for record in self:
             record.l10n_din5008_addresses = data = []
-            if record.partner_shipping_id == record.partner_id:
-                data.append((_("Invoicing and Shipping Address:"), record.partner_shipping_id))
-            elif record.move_type in ("in_invoice", "in_refund") or not record.partner_shipping_id:
-                data.append((_("Invoicing and Shipping Address:"), record.partner_id))
-            else:
-                data.append((_("Shipping Address:"), record.partner_shipping_id))
-                data.append((_("Invoicing Address:"), record.partner_id))
+            commercial_partner = record.partner_id.commercial_partner_id
+            delivery_partner = record.partner_shipping_id
+            invoice_partner = record.partner_id
+
+            different_partner_count = len({partner.id for partner in [commercial_partner, delivery_partner, invoice_partner] if partner})
+            if different_partner_count > 1:
```

Comment:
but thus that mean we nearly don't print `_("Invoicing and Shipping Address:")` if everything should be done at the same place.

Reply:
We print that when the commercial partner is different from either the delivery or invoicing address and that both are the same